### PR TITLE
Address minor portability issues

### PR DESCRIPTION
**Pull request by dankm**
Address some minor annoyances when building on FreeBSD (and possibly other systems):

Use $(MAKE) instead of "make" when determining the GNU Make version
Add OS specific sed for FreeBSD

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -469,17 +472,17 @@ $(CSS_DEST): node_modules $(CSS_SOURCES)
 
 .PHONY: javascripts
 javascripts:
-	echo "'make javascripts' is deprecated, please use 'make js'"
+	echo "'$(MAKE) javascripts' is deprecated, please use 'make js'"
```
Comment (lunny) [inline Makefile]: still have `make js`

Comment (dankm) [reply Makefile]: Ah man. Thanks. I'll fix that right away.

**Review by zeripath — APPROVED** [review]

File: Makefile [hunk 2]
```diff
@@ -141,7 +144,7 @@ generate-swagger:
 swagger-check: generate-swagger
 	@diff=$$(git diff '$(SWAGGER_SPEC)'); \
 	if [ -n "$$diff" ]; then \
-		echo "Please run 'make generate-swagger' and commit the result:"; \
+		echo "Please run '$(MAKE) generate-swagger' and commit the result:"; \
```
Comment (mrsdizzie) [inline Makefile]: Why use $(MAKE) for echo statements here?  Makes sense when running code but seems to unnecessarily add full path to the echo statements which seems more confusing:
```
Please run '/Applications/Xcode.app/Contents/Developer/usr/bin/make generate-swagger' and commit the result:'
```
We don't change the default $(MAKE) either so it should always be the same as make. Maybe there is some other benefit to this?

Comment (dankm) [reply Makefile]: The big issue is really the first (line 27ish), which causes the build to produce the wrong version number on FreeBSD. The others were informational and a minor annoyance at worst. I'm good with leaving only the first one in, and reverting the others if that's what people want.

Comment (mrsdizzie) [reply Makefile]: Yea that first one makes sense. I'd prefer to revert the other changes in the echo statements for the reason stated above

Comment (dankm) [reply Makefile]: Sounds good, I'll put a new commit up shortly.

**Review by mrsdizzie — APPROVED** [review]

**Pull request merged**